base/tps/src/main/java/org/dogtagpki/server/tps/rest/TPSCertService.java

```diff
@@ -162,6 +161,56 @@ public Response findCerts(String filter, String tokenID, Integer start, Integer
 
             return createOKResponse(response);
 
+        } catch (Exception e) {
+            throw new PKIException(e.getMessage());
+        }
+    }
+
+    private Response findAllCerts(String filter, Integer start, Integer size) {
+        String method = "TPSCertService:findAllCerts: ";
+        String msg = "";
+        TPSCertCollection response = new TPSCertCollection();
+
+        logger.debug("TPSCertService.findAllCerts({}, {}, {})", filter, start, size);
+
+        TPSEngine engine = TPSEngine.getInstance();
+        try {
+            TPSSubsystem subsystem = (TPSSubsystem) engine.getSubsystem(TPSSubsystem.ID);
+            TPSCertDatabase certDatabase = subsystem.getCertDatabase();
+            Iterator<TPSCertRecord> certRecs = certDatabase.findRecords(filter).iterator();
+            TokenDatabase tokenDatabase = subsystem.getTokenDatabase();
+            List<String> authorizedProfiles = getAuthorizedProfiles();
+            if (authorizedProfiles == null) {
+                msg = "authorizedProfiles null";
+                logger.debug("{}{}", method, msg);
+                throw new PKIException(method + msg);
+            }
+
+            int i = 0;
+            // skip to the start of the page
+            for (; i < start && certRecs.hasNext(); i++) certRecs.next();
```

Comment:
There is something I do not understand in the permission checking. The check `authorizedProfiles.contains(UserResource.ALL_PROFILES)` seems not related to the record and in the original method it was checked outside of the loops. Only the type part is related to the token. Is it possible to have the list of authorised tokens to add to the `findRecords` method? This would eliminate the check in the loops.

Reply:
>Another alternative is to just skip permission checking in this code, so just return all certs matching the search filter, but when the user tries to view a specific cert we verify that they have the permission.

Viewing the cert displays the same information that is in the table, so we're showing everyone everything if we do no permission checking? Unless I am misunderstanding you.

> The check authorizedProfiles.contains(UserResource.ALL_PROFILES) seems not related to the record and in the original method it was checked outside of the loops. Only the type part is related to the token.

Yeah, this time each `certRecord` could have a different token so the token checking has to be done in the loop. I move the whole check into the loop without trying to optimise it, I'll extract `authorizedProfiles.contains(UserResource.ALL_PROFILES)`

> Is it possible to have the list of authorised tokens to add to the findRecords method? This would eliminate the check in the loops.

Unsure, I will check.